### PR TITLE
Add generic musl arch headers. NFC.

### PR DESCRIPTION
**Pull request by sbc100**
This allows us to then remove some of our emscripten-specific
arch headers that are no longer needed (since they do not
differ from the generic version).

This is part of making musl updates easier.

See #7279

### CODE REVIEW

File: system/lib/libc/musl/arch/generic/bits/errno.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#define EPERM            1
```
Comment (aheejin) [inline system/lib/libc/musl/arch/generic/bits/errno.h:1]: It seems we have also errno.h in musl/arch/emscripten/bits? Do we not remove the emscripten version? The same for other added files, such as fcntl.h, fenv.h, shm.h, and termios.h.

Comment (sbc100) [reply system/lib/libc/musl/arch/generic/bits/errno.h:1]: No, the idea is that we have all the generic headers in `arch/generic`, and then we override just some of them with headers in `arch/emscirpten`.   This is how the musl header layout works.   At install time any emscripten-specific headers will overwrite the generic one.

**Comment by sbc100** [comment]
I should have mentioned that the contents of the generic directory comes directly from musl with no changes.

File: system/lib/libc/musl/arch/generic/bits/errno.h (hunk 1 above)
Comment (aheejin) [reply system/lib/libc/musl/arch/generic/bits/errno.h:1]: Thanks for the explanation!

**Review by aheejin — APPROVED** [review]

**Pull request merged**